### PR TITLE
MapImageLayerSublayerVisibilitySample: use ListView instead of discrete Checkboxes

### PR DESCRIPTION
**Pull request by alan-edi**
### Description

Modifies MapImageLayerSublayerVisibilitySample to use a ListView instead of 3 discrete Checkboxes. This makes use of the Sublayers list now being an ObservableList.

Checklist:
- [x] Set target branch to main (current release) or v.next (next release)
- [x] Request a reviewer
- [x] Assign a reviewer
- [x] Tag reviewer in comment

### CODE REVIEW

**Comment by alan-edi** [comment]
Assigning to @01smito01 for first review please

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java [hunk 1]
```diff
@@ -16,12 +16,16 @@
 
 package com.esri.samples.map_image_layer_sublayer_visibility;
 
+import com.esri.arcgisruntime.layers.ArcGISSublayer;
```
Comment (01smito01) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: Imports need a little rearranging - this one should be down w/ the arcgisruntime imports, and `javafx.util.Stringconverter` should be moved up from the arcgisruntime imports to the bottom of the javafx imports.

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java [hunk 2]
```diff
@@ -71,13 +75,25 @@ public void start(Stage stage) {
...
 
-      // create checkboxes for each sublayer
-      CheckBox citiesBox = new CheckBox("Cities");
-      CheckBox continentsBox = new CheckBox("Continents");
-      CheckBox worldBox = new CheckBox("World");
-
-      controlsVBox.getChildren().addAll(citiesBox, continentsBox, worldBox);
-      controlsVBox.getChildren().forEach(c -> ((CheckBox) c).setSelected(true));
+      // create a ListView containing a CheckBox for the visibility of each sublayer
+      Label sublayersLabel = new Label("Sublayers");
```
Comment (01smito01) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: We could maybe use `var` here, but your call.

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java [hunk 3]
```diff
@@ -71,13 +75,25 @@ public void start(Stage stage) {
...
+      ListView<ArcGISSublayer> sublayersList = new ListView<>();
+      sublayersList.setCellFactory(CheckBoxListCell.forListView(
+          sublayer -> {
+            // set visibility of sublayer when checkbox is clicked
+            var visibleProperty = new SimpleBooleanProperty(sublayer.isVisible());
+            visibleProperty.addListener((observable, oldValue, newValue) -> sublayer.setVisible(newValue));
+            return visibleProperty;
+          },
+          new StringConverter<>() {
```
Comment (01smito01) [inline image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: We may want to split out this StringConverter as an inner class - there's some samples where we have it inline like here, and other ones where we split it out. I'd split it out as I think having it inline distracts from the rest of the workflow here. See `BrowseBuildingFloorsSample` for an example of splitting it out in similar-ish circumstances.

**Review by 01smito01 — APPROVED** [review]
Pretty happy with this @alan-edi, have made a few suggestions - apart from the import ordering I'll leave them up to you. The image will also need recapturing - needs to be 800*400 pixels exactly. The current one is pretty close so you can probably just resize it in preview.

File: image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java (hunk 3 above)
Comment (alan-edi) [reply image_layers/map-image-layer-sublayer-visibility/src/main/java/com/esri/samples/map_image_layer_sublayer_visibility/MapImageLayerSublayerVisibilitySample.java]: OK, I think that is better

**Review by alan-edi — COMMENTED** [review]
Thanks @01smito01, i've made all those changes, including changing the image size

**Comment by alan-edi** [comment]
Assigning to @D-R-Smith for second review please

**Review by D-R-Smith — APPROVED** [review]
@alan-edi looks fine to me

**Comment by alan-edi** [comment]
Thanks @D-R-Smith, merging

**Pull request merged**